cluster/service_registry_test.go

```diff
@@ -0,0 +1,143 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/client"
+	"go.etcd.io/etcd/embed"
+)
+
+var TestEtcdAddr string
+
+func TestMain(m *testing.M) {
+	addr, cleanup := startTestEtcd()
+	defer cleanup()
+	TestEtcdAddr = addr
+
+	os.Exit(m.Run())
+}
+
+func TestServiceRegistry_Register(t *testing.T) {
+	cleanEtcdDir(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sr, err := NewServiceRegistry(ctx, TestEtcdAddr)
+	require.NoError(t, err)
```

Comment:
Put these in a helper function that returns the service registry, context, and error?

Reply:
I don't think it'd be worth it since we'd want context cancel too